docs/refactoring/confirmation-page-structure/README.md

```diff
@@ -0,0 +1,71 @@
+   - As new component for gas selection popups is created this code [handleEditGas, handleCloseEditGas](https://github.com/MetaMask/metamask-extension/blob/develop/ui/pages/confirm-transaction-base/confirm-transaction-base.component.js#L276) can be moved to it.
+   - Convert the component into a functional components and extract out all of these functions into a hook - `handleEdit`, `handleCancelAll`, `handleCancel`, `handleSubmit`, `handleSetApprovalForAll`, ``.
+3. ### [confirm-page-container](https://github.com/MetaMask/metamask-extension/tree/03ccc5366cf31c9fa0fedc2fac533ebc64e6f2b4/ui/components/app/confirm-page-container) cleanup:
+   - Move code specific to transaction to their confirmation component, for instance code related to `ApproveForAll` should be moved to `/pages/confirm-approve`, code related to `hideTitle` can be moved to `/pages/confirm-contract-interaction` etc.
+   - All header related code [here](https://github.com/MetaMask/metamask-extension/blob/03ccc5366cf31c9fa0fedc2fac533ebc64e6f2b4/ui/components/app/confirm-page-container/confirm-page-container.component.js#L191) should be moved to [confirm-page-container-header](https://github.com/MetaMask/metamask-extension/tree/03ccc5366cf31c9fa0fedc2fac533ebc64e6f2b4/ui/components/app/confirm-page-container/confirm-page-container-header)
+   - All warnings related code can be moved to a new child component.
+   - Props passing to `confirm-page-component` should be reduced. A lot of passed props like `origin`, `supportEIP1559` can be obtained directly using selectors. Props passing from `confirm-page-container` down to its child components should also be reduced.
+4. ### Gas polling:
+   Gas polling related code in `/pages/confirm-transaction` can be moved into a hook and included in `pages/confirm-transaction-base`, `/app/token-allowance` as only those confirmation pages need gas estimates.
+5. ### Refactor gasFeeInput hook
+    [GasFeeInput](https://github.com/MetaMask/metamask-extension/tree/develop/ui/hooks/gasFeeInput) is huge hook exporting about 18 different transaction related values and functions. We need to break this down into smaller hooks and utility functions.
+    The main use o fthis hook is inside [GasFeeContext](https://github.com/MetaMask/metamask-extension/blob/develop/ui/contexts/gasFee.js). Transaction pages are rendered for a transaction, this transaction can is set in this context so that we can avoid passing it over to multiple levels of child components.
+    - Some of the values like `gasFeeEstimates`, `isNetworkBusy` can be directly obtained from selectors and should be removed from the context.
+    - Hooks like `useGasPriceInput`, `useMaxFeePerGasInput`, `useMaxPriorityFeePerGasInput`, etc should be removed from this huge hook and used independent of this.
+Note: This document does not cover signature request pages which are covered separately.
```

Comment:
for some reason, this line is being combined with the last bulletpoint. Maybe.. 

```suggestion

**Note:** This document **does not cover signature request pages** which are covered separately.
```

current:
![Screen Shot 2023-02-15 at 8 03 17 PM](https://user-images.githubusercontent.com/20778143/219035473-b56c05e9-cf88-4521-aaed-ff3a6ee5afb2.png)